client/src/entrypoints/admin/draftail.js

```diff
@@ -42,3 +42,14 @@ const entityTypes = [
 ];
 
 entityTypes.forEach((type) => draftail.registerPlugin(type, 'entityTypes'));
+
+/**
+ * Initialize a Draftail editor on a given element when the w-draftail:init event is fired.
+ */
+document.addEventListener('w-draftail:init', ({ detail = {}, target }) => {
+  const id = target.id;
+  if (!id) {
+    throw new Error('w-draftail:init event must have a target with an id');
+  }
+  window.draftail.initEditor(`#${id}`, detail, document.currentScript);
+});
```

Comment:
OK, it turns out this is a nightmare to test as errors thrown in event listeners are unhandled errors.
Our goal here is to inform developers they are using this wrong so let's adjust to be a console error.

Additionally, let's just add a defensive `target` as an empty object, just in case. For the error, a minor reword to use a sentence structure.

```suggestion

/**
 * Initialize a Draftail editor on a given element when the w-draftail:init event is fired.
 */
document.addEventListener('w-draftail:init', ({ detail = {}, target = {} }) => {
  const id = target.id;

  if (!id) {
    // eslint-disable-next-line no-console
    console.error('`w-draftail:init` event must have a target with an id.');
    return;
  }

  window.draftail.initEditor(`#${id}`, detail, document.currentScript);
});
```